Makefile

```diff
@@ -1,9 +1,27 @@
+SHELL := bash
+.SHELLFLAGS := -eu -o pipefail -c
+.DELETE_ON_ERROR:
+
+# add easyjson source file targets here
+easyjson_sources := basm/dto.go x/xbasm/parse.go
+easyjson_generated := $(easyjson_sources:.go=_easyjson.go)
+
+# Rule to generate *_easyjson.go files
+%_easyjson.go: %.go
+	@easyjson $<
+
+# Generate all *_easyjson.go files
+generate: $(easyjson_generated)
+
+.PHONY: lint
 lint:
 	golangci-lint run --config golangci.yaml
 
-pre-pr: lint
-	$(MAKE) -C ./example clean run
+pre-pr: lint generate
+	$(MAKE) clean
+	$(MAKE) -C ./example run
 
-generate:
-	easyjson --all basm/dto.go
-	easyjson x/xbasm/parse.go
+.PHONY: clean
+clean:
+	# - ignore errors due to missing files
+	-$(MAKE) -C ./example clean
```

Comment:
this breaks if the files intended to be deleted are not there. alternatively, we can use `rm -f` to remove the files. I was worried about it deleting something the user did not mean to delete, though it does seem pretty standard.

Reply:
Was the "this breaks" comment made before you added the `-` ignore specifier to the top-level make file? Because from what I can tell it works fine---if the example makefile fails it ignores the error.

We should put the `-` flag in both makefiles

Reply:
yep, my bad, my comment was unclear given the fact that the line I commented on does not in fact break (because it uses the `-` ignore errors syntax). I will add the `-` to the makefile in `/example`. I think it would be ok to remove it here so the "ignore errors" directive is local to the reason it is ok to ignore errors in this case.